web-api/src/storage/postgres.rs

```diff
@@ -421,22 +421,32 @@ impl Database {
         let mut tx = self.begin().await?;
 
         let mut ingestable = ids.into_iter().collect::<HashSet<_>>();
-        let (unchanged, removed) = sqlx::query_as::<_, DocumentId>(
+        let mut unchanged = HashSet::new();
+        let mut removed = HashSet::new();
+
+        let mut rows = sqlx::query_as::<_, (DocumentId,)>(
             "SELECT document_id
             FROM document
             WHERE is_candidate;",
         )
-        .fetch_all(&mut tx)
-        .await?
-        .into_iter()
-        .partition::<Vec<_>, _>(|id| ingestable.remove(id));
+        .fetch(&mut tx);
+        while let Some((document_id,)) = rows.next().await.transpose()? {
+            if ingestable.contains(&document_id) {
+                unchanged.insert(document_id);
+            } else {
+                removed.insert(document_id);
+            }
+        }
+        drop(rows);
+
+        ingestable.retain(|id| !unchanged.contains(*id));
 
         let mut builder = QueryBuilder::new(
             "UPDATE document
             SET is_candidate = FALSE
             WHERE document_id IN ",
         );
-        let mut chunks = IterAsTuple::chunks(Self::BIND_LIMIT, removed);
+        let mut chunks = IterAsTuple::chunks(Self::BIND_LIMIT, removed.iter());
```

Comment:
:art: we could leave the iteration to `IntoIterator`
```suggestion
        let mut chunks = IterAsTuple::chunks(Self::BIND_LIMIT, &removed);
```